src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -279,7 +279,8 @@ public void validateAppEngineJavaComponents()
       throws AppEngineJavaComponentsNotInstalledException {
     if (!Files.isDirectory(getJavaAppEngineSdkPath())) {
       throw new AppEngineJavaComponentsNotInstalledException(
-          "Validation Error: Java App Engine components not installed");
+          "Validation Error: Java App Engine components not installed."
+              + " Fix by running 'gcloud components install app-engine-java' on command-line.");
```

Comment:
Didn't we say we wouldn't suggest fixes here in the exception message, but instead let the clients determine how to handle it? One reason would be if clients auto download the component, it might not make sense to have this here.